src/main/java/cookbuddy/commons/util/ImageUtil.java

```diff
@@ -0,0 +1,194 @@
+        this.PLACEHOLDER_IMAGE_STREAM = getResourceAsInputStream(PLACEHOLDER_IMAGE_PATH_STRING);
+        this.PLACEHOLDER_IMAGE = getImage(PLACEHOLDER_IMAGE_STREAM);
+    }
+
+    public static ImageUtil imageUtil() {
+        return new ImageUtil();
+    }
+
+    /**
+     * Returns a {@link BufferedImage} from the given {@code Path}
+     *
+     * @param path An image file path
+     * @return An image object
+     */
+    public BufferedImage readImage(Path path) {
+        try {
+            return ImageIO.read(path.toFile());
+        } catch (IOException e) {
+            logger.warning(MESSAGE_CONSTRAINTS);
+            return this.PLACEHOLDER_IMAGE;
+        }
+    }
+
+    /**
+     * Returns an {@link InputStream} from the given {@code image}, by invoking a
+     * {@link ByteArrayOutputStream} and copying the data out.
+     *
+     * @param image An image object
+     * @return A data object representing that image
+     */
+    public InputStream getImageInputStream(BufferedImage image) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream() {
+            @Override
+            public synchronized byte[] toByteArray() {
+                return this.buf;
+            }
+        };
+
+        try {
+            ImageIO.write(image, "png", baos);
+        } catch (IOException e) {
+            logger.warning("An error has occurred. Using placeholder image instead.");
+            return PLACEHOLDER_IMAGE_STREAM;
+        }
+        return new BufferedInputStream(new ByteArrayInputStream(baos.toByteArray()));
+    }
+
+    /**
+     * Returns a {@link BufferedImage} from the given {@code imageInputStream}
+     *
+     * @param imageInputStream A data object representing an image.
+     * @return An image object.
+     */
+    public BufferedImage getImage(InputStream imageInputStream) {
+        try {
+            return ImageIO.read(imageInputStream);
+        } catch (IOException e) {
+            logger.warning("An error has occurred. Using placeholder image instead.");
+            return PLACEHOLDER_IMAGE;
+        }
+    }
+
+    /**
+     * Returns a {@link BufferedImage} from the specified {@code url}.
+     *
+     * @param url The {@link URL} that the image is located at.
+     * @return The image at the URL.
+     */
+    public BufferedImage getImage(URL url) {
+        try {
+            return ImageIO.read(url);
+        } catch (IOException e) {
+            logger.warning("An error has occurred. Using placeholder image instead.");
+            return PLACEHOLDER_IMAGE;
+        }
+    }
+
+    /**
+     * Compares two {@link BufferedImage}s, specified by {@code i1} and {@code i2},
+     * pixel-by-pixel, to determine if they are the same image.
+     * <p>
+     * Expensive O(n^2) operation; use sparingly.
+     *
+     * @return {@code true} if {@code i1}'s pixels are all equal to those of
+     *         {@code i2}'s.
+     */
+    public boolean isSameImage(BufferedImage i1, BufferedImage i2) {
```

Comment:
What's the purpose of checking if 2 images are the same?

Reply:
It is referenced by `Photograph.equals()`, because proper image equality is a pixel-by-pixel comparison.